### DISCUSSION TITLE
Quarto book, Chapters-part-chapters, can I create a third directories?

### DISCUSSION THREAD

**Discussion by DrJasonLiu** [quarto-dev/quarto-cli#2504] [Q&A]
<img width="368" alt="image" src="https://user-images.githubusercontent.com/72745009/191069065-a5627678-efcd-4a77-a5d2-303525a24888.png">
In Parts & Appendices in Quarto book, we use chapters-part-chapters to create directories, can I further create a third directory?
Like this…
<img width="228" alt="image" src="https://user-images.githubusercontent.com/72745009/191069446-8eada1a7-5cad-4d90-b037-26c4d9a4b6f3.png">

**Comment by dragonstyle** [comment]
Right now, there isn't a third level for books (e.g. chapters composed of sections, one per file). This is something we'd like to add support for in a future update, but we don't currently have a timeline for when that might happen.

**Reply by donboyd5** [reply]
One small consideration, when you have a chance to do this:

- It would (I think) be nice if you are able to implement subsections of almost arbitrary depth, each in a separate file, and each with its own heading levels

- Currently, to achieve a similar effect, I am using the shortcode "include" approach described in #2400 -- each file can start at whatever heading level is appropriate - 2, 3, 4, or whatever

- However, I have noticed one, and maybe two issues that arise when I do this:

- 1) If you insert multiple include files from within a single master qmd file, they cannot have duplicate labels. For example, often I have a libraries chunk and a load data chunk in each include file so that I can test it independently. (Perhaps there are better ways - it would be great to have a "best practices" documentation for large books with complex hierarchies at some point. Right now I just stumble around until something works.) I prefer this chunks to be labeled so that it is clear what they are in RStudio's document outline. And I'd like to use the same label in multiple files (e.g., a "libraries" label), as I can with each qmd file that is listed in _quarto.yml. But the shortcode include method requires all labeled chunks, across all files included by a single qmd file, to be unique. So I have to use labels like "libraries_fileA", "libraries_fileB", or something like that, or else not label the chunks.

- 2) This may or may not be an issue, but I have found in one large complex project I am working on that I get the following warning when I render the project with included files: "Warning: diff of engine output timed out. No source lines will be available." This may or may not be a problem - I might be doing something wrong, and I am using the latest numbered version of quarto on github (v1.2.157 as of this writing), so I realize I am doing this at my own risk, but I thought I'd mention it nonetheless.

**Reply by mcanouil** [reply]
Regarding the duplicated code cells labels, this is an inherited behaviour from `knitr`/`rmarkdown` as described <https://bookdown.org/yihui/rmarkdown-cookbook/duplicate-label.html> (with how to allow that).

**Comment by donboyd5** [comment]
Thank you very much. I will investigate this.

Don

**Reply by ShKlinkenberg** [reply]
I recently migrated a bookdown book to quarto.

https://shklinkenberg.github.io/Statistical-Inference/index.html

But this functionality falls short in quarto. We have many levels to the chapters. The maintenance of the text and UX for the end user is sub optimal. 

The readability of the TOC would be greatly improved if we could not just add more levels, but also the seperate documents for sub chapters.

**Reply by cderv** [reply]
> I recently migrated a bookdown book to quarto.

What is the feature you found missing from bookdown ?

**Reply by ShKlinkenberg** [reply]
Hi @cderv,

I'm essentially missing two options. First is the management of multiple files within one chapter. My book goes quite a few levels deep in the header structure, and managing the different topics in separate files would be more manageable. I know this can be solve by using includes, but this feels like a hack. And with many levels it is hard to keep the overview. There is also some vagueness in how chapters are hierarchically structured. I had to find out through some trails that an h2 ## does not ensure second level header in the book structure.

I would expect

`book:
  chapters:
    -  1-1-chapter-section.qmd (# chapter 1 \n ## subsection 1.1)
    -  1-2-chapter-section.qmd (## subsection 1.2)`

To at least be rendered in one chapter. Now the latter one just becomes chapter 2, disregarding the fact that there are no # h1 in the document. So, the H1 is now implicitly determined based on the files, and not on what I wan't as an author. If you really want each seperate document to have a # h1 header, a different option could also be that you could set the rendered level in the yaml depth. Though that does feel weird to me.

 `book:
  chapters:
    -  1-0-chapter-section.qmd (# chapter 1)
    -  1-1-chapter-section.qmd (# subsection 1.1)
      -  1-2-chapter-section.qmd (# subsection 1.2)`

The other thing is related to the UX of of the chapter levels. Just one level in the right side bar does not give the reader enough overview on what to find where. 

What I would find useful is if we could set the TOC depth for both the book sidebar and the page navigation separately. So, TOC-SIDEBAR: 3 would show chapter 1, 1.1 and 1.1.1 in the left sidebar.
